### PR TITLE
refactor: rework samples sidebar directory to use typescript

### PR DESCRIPTION
**Pull request by raastinn**
## Compatiblity

Any changes that require a newer version of the API are considered breaking
changes.

Common breaking changes:
 - Making a request to a new API endpoint that will return `403` or `404` in older versions of the API.
 - No longer sending request fields required by previous versions of the API 
 - Requiring new fields from an API endpoint that will not be present in
   responses from older versions of the API
 - Breaking backwards compatibility with old response data shapes

Choose one:
 - [x] The changes do not break compatibility
 - [ ] The changes potentially break compatibility

## Pre-Review Checklist
 - [X] All changes are tested
 - [X] All touched code documentation is updated
 - [x] All tests pass in your local environment
 - [x] Deepsource issues have been reviewed and addressed
 - [X] Debugging logging and commented out code has been removed

 
## Screenshots:
_Only required for visual changes_.

### CODE REVIEW

File: src/js/samples/components/Sidebar/SampleSidebarSelector.tsx [hunk 1]
```diff
@@ -23,33 +24,53 @@ export const SampleSidebarSelectorButton = styled.div`
...
+        const { id } = item;
+
         return (
             <SampleSidebarSelectorItem
-                key={result.id}
-                selected={selectedItems.includes(result.id)}
-                partiallySelected={partiallySelectedItems.includes(result.id)}
-                {...result}
+                key={item.id}
+                selected={selectedItems.includes(id as never)}
```
Comment (ReeceHoffmann) [inline src/js/samples/components/Sidebar/SampleSidebarSelector.tsx]: Hmm why the usage of `never`? Either the check or typing doesn't make much sense here

**Comment by ReeceHoffmann** [comment]
Current status? tsc check is still looks like it is having problems

File: src/js/samples/components/Create/CreateSample.tsx [hunk 2]
```diff
@@ -219,7 +219,7 @@ export default function CreateSample() {
                     values,
                 }: {
                     errors: FormikErrors<formValues>;
-                    setFieldValue: (field: string, value: string) => void;
+                    setFieldValue: (field: string, value: string | string[] | number[]) => void;
```
Comment (ReeceHoffmann) [inline src/js/samples/components/Create/CreateSample.tsx:222]:
```suggestion
                    setFieldValue: (field: string, value:  string[] | number[]) => void;
```
Why allow a single string on it's own?

File: src/js/samples/components/Sidebar/SampleSidebarSelector.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    manageLink,
+}: SampleSidebarSelectorProps) {
+    const [results, term, setTerm] = useFuse(sampleItems, ["name"], []);
+    const sampleItemComponents = results.map((item: Label | SubtractionShortlist) => (
+        <SampleSidebarSelectorItem
+            key={item.id}
+            selected={
+                selectionType === "labels"
+                    ? (selectedItems as number[]).includes(item.id as number)
+                    : (selectedItems as string[]).includes(item.id as string)
```
Comment (ReeceHoffmann) [inline src/js/samples/components/Sidebar/SampleSidebarSelector.tsx]: I would give this a rework using generics. You should be able to declare a type such that is either a string or number but only one of the two. As written this introduces an`if` into the runtime,  while the types are removed at compile time.

**Review by ReeceHoffmann — CHANGES_REQUESTED** [review]

File: src/js/samples/components/Create/CreateSample.tsx (hunk 2 above)
Comment (raastinn) [reply src/js/samples/components/Create/CreateSample.tsx:222]: Some of the other form values like `group` use strings

**Review by ReeceHoffmann — APPROVED** [review]

**Pull request merged**